components/catalogs/back/src/Application/Mapping/ProductMapper.php

```diff
@@ -57,15 +62,17 @@ public function getMappedProduct(
                 $productMapping[$targetCode]['source'] !== null &&
                 \array_key_exists($productMapping[$targetCode]['source'], $attributeTypeBySource)) {
                 try {
-                    $sourceValue = $this->attributeValueExtractorRegistry->extract(
+                    $productValueExtractorRegistry = $this->valueExtractorRegistry->build($target['type'], $target['format'] ?? null);
+
+                    $sourceValue = $productValueExtractorRegistry->extract(
                         $product,
                         $productMapping[$targetCode]['source'],
                         $attributeTypeBySource[$productMapping[$targetCode]['source']],
```

Comment:
```suggestion
                        $attributeTypeBySource[$productMapping[$targetCode]['source']] ?? $productMapping[$targetCode]['source'],
```

If it's not a known attribute, you can fallback on the `source` itself, it will output `family` for example.